hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java

```diff
@@ -249,12 +240,13 @@ private BlockingWorker(TaskletTracker tracker, CountDownLatch startedLatch) {
         @Override
         public void run() {
             final ClassLoader clBackup = currentThread().getContextClassLoader();
-            final Tasklet t = tracker.tasklet;
+            Tasklet t = tracker.tasklet;
             currentThread().setContextClassLoader(tracker.jobClassLoader);
             IdleStrategy idlerLocal = idlerNonCooperative;
 
             try {
                 blockingWorkerCount.incrementAndGet();
+                MetricsImpl.container().setContext(t.getMetricsContext());
```

Comment:
minor, but you can assign MetricsImpl.container() to a variable instead of calling it twice